staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -216,6 +217,9 @@ func (c *dynamicResourceClient) DeleteCollection(opts *metav1.DeleteOptions, lis
 }
 
 func (c *dynamicResourceClient) Get(name string, opts metav1.GetOptions, subresources ...string) (*unstructured.Unstructured, error) {
+	if name == "" {
+		return nil, fmt.Errorf("name is required")
```

Comment:
this validation should also be applied to `Patch`, `Delete`, `Update`, `UpdateStatus`.

Reply:
I updated `Patch` and `Delete`. The other ones didn't have a `name` parameter to validate.

Reply:
update/updatestatus/patch can use `accessor.GetName()`

create should also validate name `if len(subresources) > 0`